src/utils/src/ExponentialBackoffRetryStrategy.c

```diff
@@ -232,6 +232,20 @@ STATUS getExponentialBackoffRetryStrategyWaitTime(PRetryStrategy pRetryStrategy,
     return retStatus;
 }
 
+STATUS getExponentialBackoffRetryCount(PRetryStrategy pRetryStrategy, PUINT32 retryCount) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PExponentialBackoffRetryStrategyState pRetryState = NULL;
+    pRetryState = TO_EXPONENTIAL_BACKOFF_STATE(pRetryStrategy);
+    CHK_STATUS(pRetryStrategy != NULL && pRetryState != NULL, STATUS_NULL_ARG);
```

Comment:
We should use CHK here not CHK_STATUS, CHK evaluates the condition in an `if` but CHK_STATUS applies `STATUS_FAILED` its meant to check value that's a STATUS but here it's a boolean.